plugins/modules/mso_schema_site_service_graph.py

```diff
@@ -225,29 +280,77 @@ def main():
             )
 
         if devices is not None:
-            service_node_type_names_from_template = [type.get("name") for type in service_node_types_from_template]
+            query_device_data = mso.lookup_service_node_device(site_id, tenant, device_name=None, service_node_type=None)
             for index, device in enumerate(devices):
-                template_node_type = service_node_type_names_from_template[index]
-                apic_type = "OTHERS"
-                if template_node_type == "firewall":
-                    apic_type = "FW"
-                elif template_node_type == "load-balancer":
-                    apic_type = "ADC"
-                query_device_data = mso.lookup_service_node_device(site_id, tenant, device.get("name"), apic_type)
-                devices_payload.append(
-                    dict(
-                        device=dict(
-                            dn=query_device_data.get("dn"),
-                            funcTyp=query_device_data.get("funcType"),
-                        ),
-                        serviceNodeRef=dict(
-                            serviceNodeName=template_node_type,
-                            serviceGraphName=service_graph,
-                            templateName=template,
-                            schemaId=schema_id,
-                        ),
-                    ),
-                )
+                device_payload = dict()
+                if query_device_data:
```

Comment:
So when query_device_data is None and empty dict will still be added to the payload? Is there a reason for this?